### PR TITLE
feat: MNIST Classifier with <10k params, 99.04% accuracy

### PR DESCRIPTION
**Pull request by dkstlzk**
## 📝 What does this Pull Request do?

Adds a Dockerized PyTorch-based MNIST classifier with <10k params and >98.5% accuracy.
Integrates wandb logging, Hydra configs, and Ray Tune for experiment management.

## 🔧 Type of Change

- [ ] 🐛 Bug fix (fixes something that wasn't working)
- [x] ✨ New feature (adds new functionality)
- [ ] 💅 Style change (colors, fonts, layout - no functionality change)
- [ ] 📚 Documentation update (README, guides, comments)
- [ ] 🧪 Tests (adding or fixing tests)
- [ ] 🔧 Refactoring (improving code without changing functionality)

## 🎯 Which part of the project does this affect?

- [x] 🧠 AI/ML Models (AI components in `ai/`)
- [ ] 🔧 Backend (Server logic in `backend/`)
- [ ] 🌐 Frontend (CLI & React.js in `frontend/`)
- [ ] 🌐 Network Layer (P2P networking in `network/`)
- [ ] � Shared Utilities (Common code in `shared/`)
- [ ] 📚 Documentation
- [ ] 🚀 CI/CD (GitHub workflows)

## 🧪 How did you test this?

Ran the training inside a Docker container with W&B logging enabled.
Verified training metrics and logs on the W&B dashboard successfully.

raytune
train_run
train_run2
wandb_raytune

## ✅ Checklist

- [x] 🔍 I tested my changes and they work
- [x] 💬 My commit messages follow the format (like `feat: add login button`)
- [x] 🌿 My branch has a descriptive name (like `add-login-button` or `fix-spelling`)
- [x] 📝 I have correctly updated the workflows for any new tech stack that I used
- [x] 📖 I read the [Contributing Guide](../CONTRIBUTING.md) if this is my first PR
- [x] 🤔 I would be comfortable explaining my changes to someone else

### CODE REVIEW

File: tasks/mnist_classifier/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+torch==2.3.0+cpu
```
Comment (Laksh-Mendpara) [inline tasks/mnist_classifier/requirements.txt:1]: are you sure,  using cpu version of pytorch?

File: tasks/mnist_classifier/src/raytune.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    device= torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+    transform= transforms.Compose([
+        transforms.RandomRotation(10),
+        transforms.ToTensor(),
+        transforms.Normalize((0.1307,), (0.3081,))
+    ])
+
+    train_dataset= datasets.MNIST(root="data", train=True, download=True, transform=transform)
+    test_dataset= datasets.MNIST(root="data", train=False, download=True, transform=transform)
```
Comment (Laksh-Mendpara) [inline tasks/mnist_classifier/src/raytune.py:23]: the path of root of data should be included in .gitignore

**Review by Laksh-Mendpara — APPROVED** [review]
cool
